### PR TITLE
Add command to reset the commit author from the commits panel.

### PR DESCRIPTION
**Pull request by JensPfeifle**
First of all I want to say thanks for the great tool!

This adds a command for resetting the author of a single commit, as described in #1871. 

One thing I'm no so not sure about is the default keybinding, to `a`. Any good alternatives? 

I haven't been able to run the test suite yet (doesn't seem to play nice with the docker image?), but am happy to spend some more time to add some tests if required. Also, feel free to close this if it isn't something you want to add. 

There's also a version which allows the user to edit the author name in [this branch](https://github.com/JensPfeifle/lazygit/tree/change_author_name), but this is simpler and should cover the general use-case.

### CODE REVIEW

File: pkg/commands/git_commands/commit.go [hunk 1]
```diff
@@ -22,6 +22,11 @@ func (self *CommitCommands) RewordLastCommit(message string) error {
 	return self.cmd.New("git commit --allow-empty --amend --only -m " + self.cmd.Quote(message)).Run()
 }
 
+// Reset the author of the topmost commit.
```
Comment (mark2185) [inline pkg/commands/git_commands/commit.go:25]: The convention is to start the comment with the name of the element it describes, in this case that would be `ResetAuthor reset the author of the topmost commit`

File: pkg/config/user_config.go [hunk 2]
```diff
@@ -510,6 +511,7 @@ func GetDefaultConfig() *UserConfig {
 				MoveDownCommit:                 "<c-j>",
 				MoveUpCommit:                   "<c-k>",
 				AmendToCommit:                  "A",
+				ResetCommitAuthor:              "a",
```
Comment (jesseduffield) [inline pkg/config/user_config.go:514]: I can't think of any better option than this keybinding. Although 'a' typically means 'select everything in this panel', it doesn't make any sense to select all commits. And I see the value in having 'A' and 'a' both relating to amending, with the former being for amending file changes, and the latter being for amending metadata like the author.

For what it's worth in future I want this keybinding to open a menu with an additional option to edit the author explicitly, but we don't need to do that in this PR.

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Awesome work, left some feedback

**Comment by jesseduffield** [comment]
@JensPfeifle as for the test suite, do you mean like running `go test ./...` or are you talking about the integration tests. The guide for integration tests is [here](https://github.com/jesseduffield/lazygit/blob/master/docs/Integration_Tests.md). I think this would be a good case for an integration test where we start with a commit and then open up lazygit and amend the author to the current author, but don't stress if you can't get it working on your end.

File: pkg/i18n/english.go [hunk 3]
```diff
@@ -698,6 +701,8 @@ func EnglishTranslationSet() TranslationSet {
 		LcMoveUpCommit:                      "move commit up one",
 		LcEditCommit:                        "edit commit",
 		LcAmendToCommit:                     "amend commit with staged changes",
+		LcResetCommitAuthor:                 "reset commit author",
+		SureResetCommitAuthor:               "Are you sure you want to reset the author of this commit?",
```
[the remark this answers is not in the document]
Comment (JensPfeifle) [reply pkg/i18n/english.go:705]: Good point - how about "The author field of this commit will be updated to match the configured user. This also renews the author timestamp. Continue?"

**Comment by JensPfeifle** [comment]
Thanks for the feedback!
I was having trouble with the integration tests, but got it today and managed to record a simple test.

File: pkg/i18n/english.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jesseduffield) [reply pkg/i18n/english.go:705]: Yep that's perfect

**Comment by jesseduffield** [comment]
Great work on the integration test. Heads up, I've pushed a commit that changes how we generate the snapshot of the resultant repo to include the author name and email in the git log result. Previously we had the chance of a false positive because we weren't checking against that.

**Comment by jesseduffield** [comment]
Looks like we needed to rebase onto master to get that cheatsheet thing working (now that the Japanese language has been added). I'm pushing some commits on your behalf to get this merged because the next step (allowing an author to be set) should be fairly straightforward and I'm gonna make a contributor PR guide from it.

**Comment by jesseduffield** [comment]
Merged. Great work @JensPfeifle

**Comment by JensPfeifle** [comment]
Great, thanks!

**Pull request merged**